### ISSUE TITLE
perl 5.36 cpan | Config::Any fails tests 

### ISSUE THREAD

**Issue by shawnlaffan** [StrawberryPerl/Perl-Dist-Strawberry#73]
Config::Any is needed by DBIx::Class but is failing one of its test files.

I can replicate the failures on a local machine using the same build of strawberry perl 5.36.  

Adding a ```use Config::Any::Perl``` to the top of the test file gets it to pass.  

No other build on CPAN testers has this issue which makes me suspect it is to do with the 5.36 build.  

```
--> Working on Config::Any
Fetching https://cpan.metacpan.org/authors/id/H/HA/HAARG/Config-Any-0.32.tar.gz ... OK
Configuring Config-Any-0.32 ... Checking if your kit is complete...
Looks good
Generating a gmake-style Makefile
Writing Makefile for Config::Any
Writing MYMETA.yml and MYMETA.json
Checking dependencies from MYMETA.json ...
Checking if you have Test::More 0 ... Yes (1.302193)
Checking if you have Module::Pluggable::Object 3.6 ... Yes (5.2)
OK
Building and testing Config-Any-0.32 ... cp lib/Config/Any/Base.pm blib\lib\Config\Any\Base.pm
cp lib/Config/Any/Perl.pm blib\lib\Config\Any\Perl.pm
cp lib/Config/Any/JSON.pm blib\lib\Config\Any\JSON.pm
cp lib/Config/Any/General.pm blib\lib\Config\Any\General.pm
cp lib/Config/Any/INI.pm blib\lib\Config\Any\INI.pm
cp lib/Config/Any.pm blib\lib\Config\Any.pm
cp lib/Config/Any/YAML.pm blib\lib\Config\Any\YAML.pm
cp lib/Config/Any/XML.pm blib\lib\Config\Any\XML.pm
"C:\strawberry\perl\bin\perl.exe" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0,
 'blib\lib', 'blib\arch')" t/*.t
#
# Optional Prereq Versions:
#   Config::General           missing
#   Config::Tiny              missing
#   Cpanel::JSON::XS          4.36
#   JSON::MaybeXS             1.004004
#   JSON::DWIW                missing
#   JSON::XS                  4.03
#   JSON::Syck                missing
#   JSON::PP                  4.16
#   JSON                      4.10
#   XML::Simple               2.25
#   XML::NamespaceSupport     1.12
#   YAML::XS                  0.86
#   YAML::Syck                missing
#   YAML                      1.30
#
t/00-report-prereqs.t .. ok
There are no loaders available for .pl files at t/10-branches.t line 55.
# Looks like your test exited with 2 just after 5.
t/10-branches.t ........
Dubious, test returned 2 (wstat 512, 0x200)
Failed 4/9 subtests
t/20-parse.t ........... ok
t/50-general.t ......... skipped: Config::General format not supported
t/51-ini.t ............. skipped: INI format not supported
t/52-json.t ............ ok
t/53-perl.t ............ ok
t/54-xml.t ............. ok
t/55-yaml.t ............ ok
t/61-features.t ........ ok
t/62-multi.t ........... ok
t/63-unsupported.t ..... ok
t/64-extfail.t ......... ok
t/65-force_plugins.t ... ok
t/perl-taint.t ......... ok

Test Summary Report
-------------------
t/10-branches.t      (Wstat: 512 (exited 2) Tests: 5 Failed: 0)
  Non-zero exit status: 2
  Parse errors: Bad plan.  You planned 9 tests but ran 5.
Files=15, Tests=128,  4 wallclock secs ( 0.06 usr +  0.05 sys =  0.11 CPU)
Result: FAIL
Failed 1/15 test programs. 0/128 subtests failed.
gmake: *** [makefile:851: test_dynamic] Error 255
! Installing Config::Any failed. See C:\spbuild\msys64\.cpanm\work\1678692157.2104\build.log for details. Retry with --force to for
ce install it.
```

**Comment by shawnlaffan** [comment]
Reported to Config::Any issue tracker:  https://rt.cpan.org/Ticket/Display.html?id=147112

**Comment by shawnlaffan** [comment]
This is due to an issue in File::Find - see https://rt.cpan.org/Transaction/Display.html?id=2464022 

It has been fixed in perl 5.38 via https://github.com/Perl/perl5/pull/20008 and will hopefully be backported for 5.36.1 and 5.34.2 (if the latter is released).  Otherwise then we can apply the patches from that PR to the build.

**Comment by shawnlaffan** [comment]
The perl PR 20008 patch is now part of the build (09df569).

Closing issue.
